scilpy/connectivity/matrix_tools.py

```diff
@@ -36,9 +37,11 @@ def compute_olo(array):
     if array.ndim != 2:
         raise ValueError('RCM can only be applied to 2D array.')
 
-    Z = hierarchy.ward(array)
+    condensed_array = squareform(array)
```

Comment:
I don't understand why squareform would help here? A connectivity array (like streamline count) is already a form of distance matrix. I think it is fine as it was, no?

What was your goal when you found and added squareform?

Reply:
When trying to solve this issue, I learned that a distance matrix is symmetric and can be condensed into a a list or vector. Similarly to a graph adjacency matrix that you transform into an adjacency list.

The error was that the matrix looked "suspiciously like an uncondensed distance matrix" when used in the optimal_leaf_ordering function. Using a symmetric (redundant) distance matrix in scipy functions gives a warning because it's suboptimal. (I suppose)

Squareform converts a uncondensed distance matrix into a condensed one and vice versa. The error disapeared after that.

Reply:
Ok but our matrices is real data, we can't change it to fix an error. Did you verify if the input/output still made sense?

Otherwise I would leave it like that.

Reply:
I'm not sure what you mean by real data. Maybe we should walk about it in person. 😄